bootstrap/004_labels.sql

```diff
@@ -134,6 +134,102 @@ BEGIN
 
 END;
 
+CREATE OR REPLACE PROCEDURE INTERNAL.UPSERT_PREDEFINED_LABEL(name text, grp text, rank number, condition text)
+    RETURNS text
+    LANGUAGE SQL
+    EXECUTE AS OWNER
+AS
+BEGIN
+    if (:name is null) then
+      return 'Name must not be null.';
+    elseif (grp is null and rank is not null) then
+      return 'Rank must only be set if Group name is also provided.';
+    elseif (grp is not null and rank is null) then
+      return 'Rank must provided if you are creating a grouped label.';
+    end if;
+    let outcome text := 'Failure validating name & condition. Please check your syntax.';
+
+    outcome := (CALL INTERNAL.VALIDATE_LABEL_CONDITION(:name, :condition));
+
+    if (outcome is not null) then
+      return outcome;
+    end if;
+
+    outcome := (CALL INTERNAL.VALIDATE_LABEL_Name(:name));
+
+    if (outcome is not null) then
+      return outcome;
+    end if;
+
+    MERGE INTO internal.predefined_labels t
+    USING (SELECT :name as name, :grp as grp, :rank as rank, :condition as condition) s
+    ON t.name = s.name
+    WHEN MATCHED THEN
+    UPDATE
+        SET t.GROUP_NAME = s.grp, t.GROUP_RANK = s.rank, t.CONDITION = s.condition, t.LABEL_MODIFIED_AT = current_timestamp()
+    WHEN NOT MATCHED THEN
+    INSERT
+        ("NAME", "GROUP_NAME", "GROUP_RANK", "LABEL_CREATED_AT", "CONDITION", "LABEL_MODIFIED_AT")
+        VALUES (s.name, s.grp, s.rank,  current_timestamp(), :condition, current_timestamp());
+    outcome := null;
+
+    return outcome;
+END;
+
+CREATE OR REPLACE PROCEDURE INTERNAL.DELETE_PREDEFINED_LABEL(name text)
+    RETURNS text
+    LANGUAGE SQL
+    EXECUTE AS OWNER
+AS
+BEGIN
+    if (:name is null) then
+      return 'Name must not be null.';
+    end if;
+
+    DELETE FROM internal.predefined_labels where name = :name;
+    return 'done';
+END;
+
+--CREATE OR REPLACE PROCEDURE INTERNAL.MIGRATE_PREDEFINED_LABELS()
+--    RETURNS text
+--    LANGUAGE SQL
+--    EXECUTE AS OWNER
+--AS
+--BEGIN
+--    MERGE INTO internal.labels t
+--    USING internal.predefined_labels s
+--    ON t.name = s.name and t.condition = s.condition
+--    WHEN MATCHED and t.LABEL_MODIFIED_AT <= s.LABEL_CREATED_AT THEN
+--    UPDATE
+--        SET t.GROUP_NAME = s.GROUP_NAME, t.GROUP_RANK = s.GROUP_RANK, t.CONDITION = s.condition, t.LABEL_MODIFIED_AT = s.LABEL_CREATED_AT;
+--
+--    let outcome text := 'done';
+--    return outcome;
+--END;
+
+CREATE OR REPLACE PROCEDURE INTERNAL.INITIALIZE_LABELS()
+    RETURNS boolean
+    LANGUAGE SQL
+    EXECUTE AS OWNER
+AS
+BEGIN
+    let labelcnt number := (SELECT COUNT(*) FROM internal.labels);
+
+    let label_inited text := '';
+    label_inited := (CALL INTERNAL.get_config('LABELS_INITED'));
+
+    if (labelcnt > 0 OR label_inited = 'YES') THEN
```

Comment:
I don't think this is the logic defined in #110 
what happens if the labels were inited by us but never changed and we ship more/improved labels?

Reply:
this logic is for initialization of user labels.

Per the slack https://sundeck-io.slack.com/archives/C03NCM06ARY/p1690243065568569, my understanding is we would do initialization for new install and old install.

New install:
  - labelCnt = 0  and LABELS_INITED is not done ==> we will call this SP

Old install:
  - LABELS_INITED is not done.    Only if there is no user created labels (labelCnt = 0) we will call this SP.

For either new install or old install, once this SP is called,  it will not INSERT new labels again.  When we ship more predefined labels, it would INSERT for new install or old install that met the above condition, but not the install who have called this SP before.